build/tests/01-base.sh

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env bash
+
+if [ -n "${VERBOSE:-}" ]; then
+    set -x
+fi
+
+source .requirements
+source build/tests/util.sh
+
+###
+#
+# user/group
+#
+###
+
+# a missing kong user can indicate that the post-install script on rpm/deb
+# platforms failed to run properly
+msg_test '"kong" user exists'
+assert_exec 0 'root' 'getent passwd kong'
+
+msg_test '"kong" group exists'
+assert_exec 0 'root' 'getent group kong'
+
+###
+#
+# files and ownership
+#
+###
+
+msg_test "/usr/local/kong exists and is owned by kong:root"
+assert_exec 0 'kong' "test -O /usr/local/kong || ( rc=\$?; stat '${path}'; exit \$rc )"
+assert_exec 0 'root' "test -G /usr/local/kong || ( rc=\$?; stat '${path}'; exit \$rc )"
+
+msg_test "/usr/local/bin/kong exists and is owned by kong:root"
+assert_exec 0 'kong' "test -O /usr/local/kong || ( rc=\$?; stat '${path}'; exit \$rc )"
+assert_exec 0 'root' "test -G /usr/local/kong || ( rc=\$?; stat '${path}'; exit \$rc )"
+
+if alpine; then
+    # we have never produced real .apk package files for alpine and thus have
+    # never depended on the kong user/group chown that happens in the
+    # postinstall script(s) for other package types
+    #
+    # if we ever do the work to support real .apk files (with read postinstall
+    # scripts), we will need to this test
+    msg_yellow 'skipping file and ownership tests on alpine'
+else
+    for path in \
+        /usr/local/bin/luarocks \
+        /usr/local/etc/luarocks/ \
+        /usr/local/lib/{lua,luarocks}/ \
+        /usr/local/openresty/ \
+        /usr/local/share/lua/; do
+        msg_test "${path} exists and is owned by kong:kong"
+        assert_exec 0 'kong' "test -O ${path} || ( rc=\$?; stat '${path}'; exit \$rc )"
+        assert_exec 0 'kong' "test -G ${path} || ( rc=\$?; stat '${path}'; exit \$rc )"
+    done
+fi
+
+msg_test 'default conf file exists and is not empty'
```

Comment:
file existents test should have already been taken care of by the manifest.
if they are not, i would suggest to add them to `filelist.txt` https://github.com/Kong/kong/blob/master/scripts/explain_manifest/filelist.txt

Reply:
> file existence test should have already been taken care of by the manifest.

These tests only incidentally test for file existence. `test -s` actually tests for file size greater than zero.

I see that the explain_manifest script does read file sizes, but I don't see that we've defined any in the manifests or the filelist. And It would probably be tedious to add a "threshold" mechanism that would give us the same "greater than zero" functionality `test -s` gives us here.

> we also have static analysis of binaries as well

It is true that these tests are pretty much "dynamic analysis" as you noted, but they do conform to environment within the container. Which is ultimately what the customer will care about.

Not to belittle this conversation, but these all feel like moot points. The goal of these changes are to try and give us the same test coverage the old images had so we can increase our confidence in the bazel-based release process. And since these tests are so small and fast to run, I don't see any drawbacks in having some overlap between these are the verify_manifest/filelist tests.

Reply:
Yes I do see the additional validation we are doing here, my concern is that we are doing two very similar things
in two seperated approaches, and it may confuse us in the future. Personally, I have imagined the next step
for the explain_manifest tool to also accept a imperative test like `Except("glob path").NonEmpty()`, `Except("nginx path").HasRpath("/usr/local/kong")` to cover more specific test cases. You are right about the runtime environment
that `dynamic analysis` is required, but it then also comes to the situation that if we would only know it work when
actually running it, then we might be putting too much variables in play that we are not even aware of.
Additionally moving forward, I would also suggest we to start write those tests in a high-level programming language so the test itself can be tested. Not saying we write bad bash, but the effort we took in debugging the scripts in the past has been more than fixing the actual issue itself.